README.md

```diff
@@ -1,92 +1,158 @@
+In order to use this suite, Ansible 2.3.2.0 is required.
+
+### Testing
+
+We use [Molecule](https://github.com/metacloud/molecule#molecule) to test Conjur's Ansible role, lookup plugin & module.
 
-A running Conjur service accessible from the host machine and target machine.
+In order to run the tests, the following are required:
 
-Role Variables
---------------
-* conjur_identity:
-  * `conjur_url`: URL of the running Conjur service
-  * `account`: Conjur account name
-  * `host_factory_token`: [Host Factory](https://developer.conjur.net/reference/services/host_factory/) token for layer enrollment
-  * `host_name`: Name of the host being conjurized.
-  * `ssl_certificate`: Public SSL certificate of Conjur endpoint
-  * `validate_certs`: yes
+* Ansible 2.3.2.0
+* Molecule 2.0.4 
 
-Dependencies
-------------
+### Dependencies
 
 None
 
-Example Playbook
-----------------
+
+## Conjur role
+
+The Conjur role can be used to configure a host with a Conjur machine identity. Through integration with Conjur, 
+the machine can then be granted least-privilege access to retrieve the secrets it needs in a secure manner. 
+Providing a node with a Conjur Identity enables permission to be dictated by Conjur policies.
+
+### Role Variables
+
+* configure-conjur-identity:
+  * `conjur_appliance_url` `*`: URL of the running Conjur service
+  * `conjur_account` `*`: Conjur account name
+  * `conjur_host_factory_token` `*`: [Host Factory](https://developer.conjur.net/reference/services/host_factory/) token for 
+  layer enrollment
+  * `conjur_host_name` `*`: Name of the host being conjurized.
+  * `conjur_ssl_certificate`: Public SSL certificate of the Conjur endpoint
+  * `conjur_validate_certs`: Boolean value to indicate if the Conjur endpoint should validate certificates
+
+The variables marked with `*` are required fields. The other variables are required for running with an HTTPS Conjur endpoint,
+but are not required if you run with an HTTP Conjur endpoint.
+
+
+### Example Playbook
 
 Configuring a remote node with a Conjur identity:
 ```yml
 - hosts: servers
   roles:
-    - role: conjur_identity
-      account: 'myorg',
-      conjur_url: 'https://conjur.myorg.com/api',
-      host_factory_token: "{{lookup('env', 'HFTOKEN')}}",
-      host_name: "{{inventory_hostname}}"
+    - role: playbook.yml
+      conjur_appliance_url: 'https://conjur.myorg.com/api',
+      conjur_account: 'myorg',
+      conjur_host_factory_token: "{{lookup('env', 'HFTOKEN')}}",
+      conjur_host_name: "{{inventory_hostname}}"
 ```
 
-Provide secret retrieval:
+## Conjur lookup plugin
+
+Conjur's `retrieve_conjur_variable` lookup plugin provides a means for retrieving secrets from Conjur for use in playbooks. 
+Note that as lookup plugins run in the Ansible host machine, the identity that will be used for retrieving secrets
+are those of the Ansible host. Thus, the Ansible host requires god like privilege, essentially read access to every secret that a remote node may need.
```

Comment:
The 'god-like privilege thing' isn't necessarily true. If you are using this method to create remote nodes, then the Ansible host only requires enough privilege for whatever it takes to create the node and enroll it in Conjur. Once the remote node exists it can fetch it's own secrets without needing the control node to have privileges on them. That is the use case where the capabilities of the lookup plugin are most valuable.

Reply:
I'm going to merge this, then your PR @Daniel-Warner-X. That should resolve this.

Reply:
@Daniel-Warner-X the lookup_plugin isn't used to create remote nodes, that's what we do in the role. The plugin is used when your remote nodes are not conjurized and you retrieve secrets using the Ansible host machine identity. Thus, it is god-like as it needs access to secrets that are needed by all of its remote nodes